spec/model/project_spec.rb

```diff
@@ -57,4 +57,25 @@
       project.soft_delete
     end
   end
+
+  describe ".default_location" do
```

Comment:
We have some unit tests that directly use the database, and others that extensively mock DB requests. I'm unsure which approach we should adopt. We encountered a similar issue at my previous company. Some colleagues believe we should mock all external components to keep the unit tests highly isolated, while others think we should use a real database. The line between when to mock or not is not clear and thin for me, and I'm uncertain about where to draw the line.
For instance, in this particular test, we mock all DB requests. I'm wondering what benefits these tests offer, aside from increasing line coverage.

At times, I struggle to decide whether to mock a class or not. I'm posing this question to gain a better understanding of the topic.

@fdr might have some insights on this topic.

Reply:
Fair point. I changed the tests to use and not mock the database. I agree that these tests cover a wider spectrum of the system under test.